RFCs/0004-add-pdk-new-transport.md

```diff
@@ -0,0 +1,49 @@
+# Add `pdk new transport` Subcommand
+
+## Summary
+
+Add a new `pdk new transport` subcommand to create a ready-to-code Resource API transport, schema and test skeleton, similar to `pdk new provider`.
+
+## Background & Assumptions
+
+- The Resource API (and through bundling Puppet 6.4+ and bolt 1.14+) support _transports_ as a means to manage remote resources and execute tasks for systems that do not follow the conventional UNIX/Windows model of on-system execution.
+
+- Building resources and tasks for this kind of system requires extra coding to facilitate the communication with those systems.
+
+- A transport schema (like a type) describes the information required to communicate with a remote target.
+
+- A transport (like a provider) implements the communication with the remote target.
+
+- See https://github.com/puppetlabs/puppet-specifications/blob/master/language/resource-api/README.md#transports for the full specification.
+
+## Motivation & Goals
+
+- PDK should provide an easy way for module authors to build advanced content.
+
+- PDK should encourage module authors to adopt good development practices, such as maintaining unit test coverage of their code.
+
+## Proposed Implementation Design
+
+Implement a new pdk subcommand, `pdk new transport <transport_name>` which will render a set of templates from `object_templates/transport_*` using the provided transport name.
```

Comment:
Are there any other relevant arguments or is it just `<transport_name>`?

Reply:
I don't think so. `pdk new provider` didn't need any either.